include/deal.II/numerics/data_out_dof_data.templates.h

```diff
@@ -242,44 +278,60 @@ namespace internal
         {
           Assert(data.x_fe_face_values.empty() == true, ExcInternalError());
 
-          const bool do_simplex =
-            std::any_of(finite_elements.begin(),
-                        finite_elements.end(),
-                        [](const auto &fe) {
-                          for (unsigned int i = 0; i < fe->size(); ++i)
-                            if ((*fe)[i].reference_cell_type() !=
-                                ReferenceCell::get_hypercube(dim))
-                              return true;
-                          return false;
-                        });
-
-          const bool do_hex =
-            std::any_of(finite_elements.begin(),
-                        finite_elements.end(),
-                        [](const auto &fe) {
-                          for (unsigned int i = 0; i < fe->size(); ++i)
-                            if ((*fe)[i].reference_cell_type() ==
-                                ReferenceCell::get_hypercube(dim))
-                              return true;
-                          return false;
-                        });
-
-          std::unique_ptr<dealii::Quadrature<dim>> quadrature1;
-          std::unique_ptr<dealii::Quadrature<dim>> quadrature2;
-
-          if (do_simplex)
+          // determine if specific quadrature rules need to set up
+          bool needs_hypercube_setup = false;
+          bool needs_simplex_setup   = false;
+          bool needs_wedge_setup     = false;
+
+          for (const auto &fe : finite_elements)
+            for (unsigned int i = 0; i < fe->size(); ++i)
+              switch ((*fe)[i].reference_cell_type())
+                {
+                  case ReferenceCell::Type::Vertex:
+                  case ReferenceCell::Type::Line:
+                  case ReferenceCell::Type::Quad:
+                  case ReferenceCell::Type::Hex:
+                    needs_hypercube_setup |= true;
+                    break;
+                  case ReferenceCell::Type::Tri:
+                  case ReferenceCell::Type::Tet:
+                    needs_simplex_setup |= true;
+                    break;
+                  case ReferenceCell::Type::Wedge:
+                    needs_wedge_setup |= true;
+                    break;
+                  default:
+                    Assert(false, ExcNotImplemented());
+                }
+
+          std::unique_ptr<dealii::Quadrature<dim>> quadrature_simplex;
+          std::unique_ptr<dealii::Quadrature<dim>> quadrature_hypercube;
+          std::unique_ptr<dealii::Quadrature<dim>> quadrature_wedge;
+
+          if (needs_simplex_setup)
```

Comment:
How hard would it be to generalize this part of the setup? This code is identical to the other one at circa line 130.

Reply:
Done.